tests/cpgrid/eclCentroid_test.cpp

```diff
@@ -0,0 +1,339 @@
+#endif
+
+
+#include <opm/grid/CpGrid.hpp>
+#include <dune/common/version.hh>
+#include <dune/grid/common/mcmgmapper.hh>
+#include <opm/input/eclipse/Deck/Deck.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/input/eclipse/Parser/Parser.hpp>
+#include <opm/grid/CpGrid.hpp>
+
+
+
+#include <sstream>
+#include <iostream>
+
+struct Fixture
+{
+    Fixture()
+    {
+        int m_argc = boost::unit_test::framework::master_test_suite().argc;
+        char** m_argv = boost::unit_test::framework::master_test_suite().argv;
+        Dune::MPIHelper::instance(m_argc, m_argv);
+        Opm::OpmLog::setupSimpleDefaultLogging();
+    }
+
+    static int rank()
+    {
+        int m_argc = boost::unit_test::framework::master_test_suite().argc;
+        char** m_argv = boost::unit_test::framework::master_test_suite().argv;
+        return Dune::MPIHelper::instance(m_argc, m_argv).rank();
+    }
+};
+
+BOOST_GLOBAL_FIXTURE(Fixture);
+
+void createEclGridCpGrid_and_checkEclCentroid(const std::string& deckString)
+{
+    Opm::Parser parser;
+    const auto deck = parser.parseString(deckString);
+
+    Dune::CpGrid grid;
+    Opm::EclipseGrid ecl_grid(deck);
+
+    grid.processEclipseFormat(&ecl_grid, nullptr, false, false, false);
+    
+    const auto& gridLeafView = grid.leafGridView();
+    Dune::CartesianIndexMapper<Dune::CpGrid> gridCartMapper =  Dune::CartesianIndexMapper<Dune::CpGrid>(grid);
+    for (const auto& element: Dune::elements(gridLeafView)){
+        const auto& elemEclCentroid = ecl_grid.getCellCenter(gridCartMapper.cartesianIndex(element.index()));
+        const auto& elemCpGridEclCentroid_Entity = grid.getEclCentroid(element);
+        const auto& elemCpGridEclCentroid_Index = grid.getEclCentroid(element.index());
+        for (int coord = 0; coord < 3; ++coord)
+        {
+            BOOST_CHECK_CLOSE(elemEclCentroid[coord],elemCpGridEclCentroid_Entity[coord] , 1e-6);
+            assert(elemCpGridEclCentroid_Entity[coord] == elemCpGridEclCentroid_Index[coord]);
+            std::cout << " From Eclipse: " << elemEclCentroid[coord]
+                      << " From CpGrid (Entity): " << elemCpGridEclCentroid_Entity[coord]
+                      << " From CpGrid (Index): " << elemCpGridEclCentroid_Index[coord]<< '\n';
+        }
+        std::cout << " CpGrid center: " << element.geometry().center() << '\n';
+        std::cout << " " << '\n';
+    }
+   
+}
+
+
+
+BOOST_AUTO_TEST_CASE(stringA)
+{
+    const std::string deckString =
+        R"( RUNSPEC
+        DIMENS
+        1  1  5 /
+        GRID
+        COORD
+        0 0 0
+        0 0 1
+        1 0 0
+        1 0 1
+        0 1 0
+        0 1 1
+        1 1 0
+        1 1 1
+        /
+        ZCORN
+        4*0
+        8*1
+        8*2
+        8*3
+        8*4
+        4*5
+        /
+        ACTNUM
+        5*1
+        /
+        PORO
+        5*0.15
+        /)";
+
+    createEclGridCpGrid_and_checkEclCentroid(deckString);
```

Comment:
Would it be possible to additionally have some checks of the form
```C++
BOOST_CHECK_CLOSE(centroid_x, expected_value_x, tolerance);
BOOST_CHECK_CLOSE(centroid_y, expected_value_y, tolerance);
BOOST_CHECK_CLOSE(centroid_z, expected_value_z, tolerance);
```
The checker function "just" prints values to the standard output stream, which means that a person will have to review the output manually and we lose a little bit of the automated aspect of unit tests.

Reply:
The printing is not necessary, I agree. However, the old line 96 was doing the type of check you suggested. I added one extra line in that spirit  "centroid_*, expected_value_*, tolerance", and removed the unnecessary print-part. 
Thanks!